src/message_interface/message_handler.rs

```diff
@@ -126,15 +119,24 @@ impl WalletMessageHandler {
                         let account = account_handle.read().await;
                         accounts.push(account.clone());
                     }
-                    Ok(ResponseType::ReadAccounts(accounts))
+                    Ok(ResponseType::Accounts(accounts))
                 })
                 .await
             }
+            #[cfg(feature = "stronghold")]
+            MessageType::RestoreBackup { source, password } => {
+                let res = convert_async_panics(|| async {
+                    self.restore_backup(source.to_path_buf(), password.to_string()).await
```

Comment:
Hm for the message_interface it would work, but if a user calls the function from Rust he would need to take care of calling `.zeroize()` himself then because it doesn't get moved then, is that really better?

Reply:
Not sure to follow, if they call the function themselves, they already have a password to zeroize. Passing a reference or a clone doesn't change the fact that they need to zeroize it.

Reply:
When `stronghold_password` is a `String` it would be like this and internally zeroized and the user doesn't have to call it manually
```Rust
    restore_manager
        .restore_backup(
            PathBuf::from("test-storage/backup_and_restore/backup.stronghold"),
            stronghold_password,
        )
        .await?;
```
If we change it to `&str`, then the user would need to call `stronghold_password.zeroize();` himself, no?
```Rust
    restore_manager
        .restore_backup(
            PathBuf::from("test-storage/backup_and_restore/backup.stronghold"),
            &stronghold_password,
        )
        .await?;

    stronghold_password.zeroize();
```

Reply:
It's because we match on &mut, Pelumi probably did that to also return the MessageType in the response, which we don't need I believe
Would be like here then https://github.com/Thoralf-M/wallet.rs/pull/2